paddle/fluid/pir/transforms/shape_optimization_pass.cc

```diff
@@ -143,6 +151,67 @@ void DebugPrintOpInfo(
   }
 }
 
+void CheckInferSymWithInferMeta(
+    pir::Operation* op,
+    pir::ShapeConstraintIRAnalysis* shape_analysis = nullptr) {
+  for (uint32_t i = 0; i < op->num_results(); ++i) {
+    const auto& res = op->result(i);
+    std::ostringstream print_stream;
+
+    // InferMeta funcs of some Ops are not corrrect now, we don't check them.
+    if (check_ops_blacklist.find(op->name() + "_" + std::to_string(i)) !=
+        check_ops_blacklist.end())
+      continue;
+
+    if (res.type().isa<paddle::dialect::DenseTensorType>()) {
+      const std::vector<int64_t>& infer_meta_shape = common::vectorize(
+          res.type().dyn_cast<paddle::dialect::DenseTensorType>().dims());
+      const std::vector<symbol::DimExpr>& infer_sym_shape =
+          shape_analysis->GetShapeOrDataForValue(res).shape();
+
+      // Check rank.
+      if (infer_meta_shape.size() != infer_sym_shape.size()) {
+        std::ostringstream print_stream;
+        print_stream << "Warning : Check InferSymbolicShape for " << op->name()
```

Comment:
op->name() 后面加上op->id()
```suggestion
        print_stream << "Warning : Check InferSymbolicShape for " << op->name() << "(op_" << op->id() << ")"
```

Reply:
done